src/engine/Map.cxx

```diff
@@ -519,16 +521,15 @@ void Map::demolishNode(const std::vector<Point> &isoCoordinates, bool updateNeig
       if (pNodeTileData && ((pNodeTileData->RequiredTiles.height > 1) || (pNodeTileData->RequiredTiles.width > 1)))
       {
         const Point origCornerPoint = node.getOrigCornerPoint(Layer::BUILDINGS);
-        const size_t origIndex = nodeIdx(origCornerPoint.x, origCornerPoint.y);
 
-        if (origIndex < mapNodes.size())
+        if (origCornerPoint.isWithinMapBoundaries())
```

Comment:
isWithinMapBoundaries should be a member of Map class, not Point.

In Point class, you remove constexpr from INVALID (it should be constexpr)

Reply:
constexpr does not compile:
```
[build] /Users/lisa/Projects/Coding/Cytopia/Cytopia/src/engine/basics/point.hxx:28:26: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
[build]   static constexpr Point INVALID() { return {-1, -1, -1, -1}; }
```

and I don't agree with moving the function to map class. it was in isomath previously as a free function. to me it feels better to have it in Point.

Reply:
Did you try to set class constructor to be constexpr? (one with for parameters)

This can be static function outside the Map class but in the file where is Map class. This will still be API of the Map not the Point. But it is up to you.

Reply:
i have no idea what that does but I've added it

Reply:
Also this leads to:
```
build] /Users/lisa/Projects/Coding/Cytopia/Cytopia/src/engine/basics/point.hxx:10:11: warning: constexpr constructor that does not initialize all members is a C++20 extension [-Wc++20-extensions]
[build] constexpr Point(int x, int y, int z) : x(x), y(y), z(z) { };
```